lib/HLSL/DxilValidation.cpp

```diff
@@ -3852,8 +3841,11 @@ static void ValidateTypeAnnotation(ValidationContext &ValCtx) {
 }
 
 static void ValidateBitcode(ValidationContext &ValCtx) {
-  if (llvm::verifyModule(ValCtx.M, &ValCtx.DiagStream())) {
+  std::string diagStr;
+  raw_string_ostream diagStream(diagStr);
+  if (llvm::verifyModule(ValCtx.M, &diagStream)) {
     ValCtx.EmitError(ValidationRule::BitcodeValid);
+    dxilutil::EmitErrorOnContext(ValCtx.M.getContext(), diagStr);
```

Comment:
Either use `diagStream.str()`, or be sure to `diagStream.flush()` before using `diagStr`, otherwise you might not have the full output flushed to the string.